app/src/main/java/com/android/unio/ui/authentication/AccountDetails.kt

```diff
@@ -70,13 +71,57 @@ import com.google.firebase.Firebase
 import com.google.firebase.auth.auth
 import kotlinx.coroutines.flow.MutableStateFlow
 
-@OptIn(ExperimentalLayoutApi::class)
 @Composable
-fun AccountDetails(
+fun AccountDetailsScreen(
     navigationAction: NavigationAction,
     userViewModel: UserViewModel,
     imageRepository: ImageRepository
 ) {
+
+  val context = LocalContext.current
+  val userId = Firebase.auth.currentUser?.uid
+
+  AccountDetailsContent(
+      navigationAction,
+      onCreateUser = { profilePictureUri, createUser ->
+        if (profilePictureUri.value == Uri.EMPTY) {
+          createUser("", userId!!)
+        } else {
+          val inputStream = context.contentResolver.openInputStream(profilePictureUri.value)
+          imageRepository.uploadImage(
+              inputStream!!,
+              "images/users/${userId}",
+              onSuccess = { createUser("images/users/${userId}", userId!!) },
```

Comment:
This hardcoded Firebase storage path could be moved to Strings.kt along with the Firestore ones

Reply:
Well noticed, I'll do just that :+1: